analysis/quantifier-comparisons/10-miQC-filtering.Rmd

```diff
@@ -0,0 +1,272 @@
+---
+title: "MiQC Exploration for ScPCA"
+author: "Ally Hawkins for CCDL"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+---
+
+In this notebook, we are interested in exploring the use of [miQC](https://github.com/greenelab/miQC) as an additional approach to filtering cells to remove any remaining cells that may not be viable or have low sequencing information. 
+
+Here, I am particularly comparing the use of miQC to using hard thresholds, in particular removing cells with greater than 20% mitochondrial content, less than 500 genes detected per cell, and less than 1000 total counts per cell. 
+
+For the most part, I followed the instructions in the [miQC vignette](https://github.com/greenelab/miQC/blob/main/vignettes/miQC.Rmd) for plotting and filtering. 
+
+I'm choosing to compare 2 single-cell samples (SCPCR000126, SCPCR000127) and 2 single-nuclei samples (SCPCR000118, SCPCR000119). We are particularly interested in how miQC will handle single-nuclei samples, as they should not contain many reads corresponding to mitochondrial genes. 
+
+## Setup
+
+```{r}
+library(magrittr)
+library(ggplot2)
+library(SingleCellExperiment)
+library(scpcaTools)
+library(gridExtra)
+```
+
+
+```{r}
+# file paths to emptyDrops filtered sces from Alevin-fry 
+base_dir <- here::here()
+cr_like_results_dir <- file.path(base_dir, "data", "results")
+
+cr_like_sce_file <- file.path(cr_like_results_dir, "alevin-fry-cr-like-em-emptydrops-200-sces.rds")
+
+cr_like_sce <- readr::read_rds(cr_like_sce_file)
+```
+
+
+
+```{r}
+# read in mito gene list
+sample_info_dir <- file.path(base_dir, "sample-info")
+mito_file <- file.path(sample_info_dir, "Homo_sapiens.GRCh38.103.mitogenes.txt")
+mito_genes <- readr::read_tsv(mito_file, col_names = "gene_id")
+mito_genes <- mito_genes %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+```
+
+
+## Plot Metrics 
+
+Before we can do any modeling and perform any filtering, let's look at the distribution of mitochondrial fraction per cell and unique genes detected per cell. 
+The assumptions of miQC rely on having a distribution of mitochondrial reads and unique genes found. 
+It uses a joint model to look at the proportion of reads mapping to mitochondrial DNA and the number of detected genes and determines the probability of a cell being compromised based on this proportion. 
+
+To start we have to calculate the per cell statistics using `addPerCellQC()`. 
+```{r}
+# add per cell qc 
+cr_like_sce <- cr_like_sce %>%
+  purrr::map(scater::addPerCellQC, subsets = list(mito = mito_genes))
+```
+
+```{r}
+cr_like_sce %>%
+  purrr::map(miQC::plotMetrics)
```

Comment:
Switching to iwalk prevents extraneous output, and we probably want some titles in these plots as well. I have not tested the code, but expecting (based on the source code) that `miQCplotMetrics()` returns a `ggplot2` object, this ought to work to add a title to each plot.
```suggestion
  purrr::iwalk(~miQC::plotMetrics(.x) + ggtitle(.y))
```